### PR TITLE
Use String interpolation or << instead plus which are faster

### PR DESCRIPTION
**Pull request by shhavel**

### CODE REVIEW

File: lib/rack/auth/digest/params.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@ def self.dequote(str) # From WEBrick::HTTPUtils
         end
 
         def self.split_header_value(str)
-          str.scan( /(\w+\=(?:"[^\"]+"|[^,]+))/n ).collect{ |v| v[0] }
+          str.scan(/\w+\=(?:"[^\"]+"|[^,]+)/)
```
Comment (gioele) [inline lib/rack/auth/digest/params.rb]: This change looks unrelated to this PR. It also changes the way headers are parsed.

Wrong chunk selected during `git add -p`? :)

Comment (shhavel) [inline lib/rack/auth/digest/params.rb]: Ok, sorry. will remove it from PR.
